src/AlgebraicGeometry/ToricVarieties/CohomologyClasses/methods.jl

```diff
@@ -97,30 +97,30 @@ function integrate(c::CohomologyClass)::QQFieldElem
         intersection_dict = _intersection_form_via_exponents(toric_variety(c))
         coeffs = coefficients(c)
         expos = exponents(c)
-        integral = 0
+        integral = zero(QQ)
         for i in 1:nrows(expos)
             if expos[i, :] in keys(intersection_dict)
                 integral += coeffs[i] * intersection_dict[expos[i, :]]
             end
         end
-        return integral
+        return integral::QQFieldElem
     end
     
     # otherwise, proceed "by hand"
     if is_trivial(c)
-        return 0
+        return zero(QQ)
     end
     poly = polynomial(c)
     dict = homogeneous_components(poly)
     elem = base_ring(parent(poly)).D([dim(toric_variety(c))])
     if !(elem in keys(dict))
-        return 0
+        return zero(QQ)
     end
     top_form = dict[elem]
     if iszero(top_form)
-        return 0
+        return zero(QQ)
     end
     n = AbstractAlgebra.leading_coefficient(top_form.f)
     m = AbstractAlgebra.leading_coefficient(polynomial(volume_form(toric_variety(c))).f)
-    return QQFieldElem(n//m)
+    return QQFieldElem(n,m)
```

Comment:
@fingolfin Unless I am mistaken, this change leads to the CI-failures in 1.6 ubuntu-latest: `no method matching QQFieldElem(::QQFieldElem, ::QQFieldElem)`.

Generally speaking, your changes look good to me. In particular, I would hope that once the above line is fixed, the tests pass again.

Reply:
Maybe `QQFieldElem(ZZ(n), ZZ(m))` might work.

Reply:
I just tested this fix, and it seems to work. So I have taken the liberty to push a corresponding fix.

(Origin: `m`, `n` are coefficients of polynomials with rational coefficients. So by construction, they are rational numbers.)